PR: #162 : Adding changes for creating weak stubs and linking the same in Makefile

kanjoe24 opened the pull request:
Adding changes for creating weak stubs and linking the same in Makefile

kanjoe24 commented:
tested this changes on HDMI-CEC halif, with following changes in test Makefile:

```makefile
cat diff_log.txt 
diff --git a/Makefile b/Makefile
index 25a3867..08901ac 100644
--- a/Makefile
+++ b/Makefile
@@ -34,7 +34,7 @@ TOP_DIR := $(ROOT_DIR)
 SRC_DIRS = $(ROOT_DIR)/src
 INC_DIRS := $(ROOT_DIR)/../include
 HAL_LIB := RCECHal
-SKELTON_SRCS := $(ROOT_DIR)/skeletons/src/hdmi_cec_driver.c
+#SKELTON_SRCS := $(ROOT_DIR)/skeletons/src/hdmi_cec_driver.c
 VCOMPONENT_SRCS := $(wildcard $(ROOT_DIR)/vcomponent/src/*.c)
 VCOMPONENT_OBJS := $(subst src,build,$(VCOMPONENT_SRCS:.c=.o))
 
@@ -42,17 +42,20 @@ ifeq ($(TARGET),)
 $(info TARGET NOT SET )
 $(info TARGET FORCED TO Linux)
 TARGET=linux
-SRC_DIRS += $(ROOT_DIR)/skeletons/src
+#SRC_DIRS += $(ROOT_DIR)/skeletons/src
 YLDFLAGS += -lpthread -lrt
 endif
 
+# Add the source files from skeletons/src directory
+BUILD_WEAK_STUBS_SRC = $(ROOT_DIR)/skeletons/src
+
 $(info TARGET [$(TARGET)])
 
 ifeq ($(TARGET),arm)
 HAL_LIB_DIR := $(ROOT_DIR)/libs
 YLDFLAGS = -Wl,-rpath,$(HAL_LIB_DIR) -L$(HAL_LIB_DIR) -l$(HAL_LIB)  -lpthread -lrt
 ifeq ("$(wildcard $(HAL_LIB_DIR)/lib$(HAL_LIB).so)","")
-SETUP_SKELETON_LIBS := skeleton
+#SETUP_SKELETON_LIBS := skeleton
 endif
 endif
 
@@ -75,6 +78,7 @@ export INC_DIRS
 export TARGET
 export TOP_DIR
 export HAL_LIB_DIR
+export BUILD_WEAK_STUBS_SRC
 
 .PHONY: clean list build skeleton vcomponent
 
diff --git a/build.sh b/build.sh
index 6b77cee..b3b26c3 100755
--- a/build.sh
+++ b/build.sh
@@ -59,7 +59,8 @@ else
     git clone ${TEST_REPO} ut-core
     check_next_revision
     cd ./ut-core
-    git checkout ${UT_CORE_PROJECT_VERSION}
+    #git checkout ${UT_CORE_PROJECT_VERSION}
+    git checkout feature/gh162-Makefile-changes-for-weak-stubs
     ./build.sh $@
     cd ..
     ./${0} $@

```

kanjoe24 commented:
Weak stubs are made in the build/${TARGET}:

 
```
 tree build/linux/
build/linux/
├── lib
│   └── libweak_stubs_libs.a
└── obj
    ├── hal_test
    ├── skeletons
    │   └── src
    │       └── hdmi_cec_driver.o
    ├── src
    │   ├── main.o
    │   ├── test_l1_hdmi_cec_driver.o
    │   ├── test_l2_hdmi_cec_sink_driver.o
    │   ├── test_l2_hdmi_cec_source_driver.o
    │   ├── test_l3_hdmi_cec_driver.o
    │   ├── test_register.o
    │   └── test_vcomponent.o
    └── ut-core
        ├── framework
        │   └── CUnit-2.1-3
        │       └── CUnit
        │           └── Sources
        │               └── Framework
        │                   ├── CUError.o
        │                   ├── MyMem.o
        │                   ├── TestDB.o
        │                   ├── TestRun.o
        │                   └── Util.o
        └── src
            ├── c_source
            │   ├── cunit_lgpl
            │   │   ├── ut_automated.o
            │   │   ├── ut_basic.o
            │   │   └── ut_console.o
            │   └── ut_cunit.o
            ├── ut_kvp_profile.o
            └── ut_main.o


```

Ulrond commented on Makefile:
```diff
@@ -171,6 +182,16 @@ checkvariantchange:
+	@$(AR) rcs $@ $^
+
+# Rule to compile .c files into .o files in the correct directory
+$(BUILD_DIR)/skelton/src/%.o: $(BUILD_WEAK_STUBS_SRC)/%.c
```
think this seems wrong

WEAK_STUBS_OUTPUT_DIR=$(BUILD_DIR)/weak_stubs/obj:

$(WEAK_STUBS_OUTPUT_DIR)/%.o: $(BUILD_WEAK_STUBS_SRC)/%.c

should it be something like this?

kanjoe24 commented:
The path for object files changed:

```
tree ut/build
ut/build
└── linux
    ├── lib
    │   └── libweak_stubs_libs.a
    └── obj
        ├── hal_test
        ├── src
        │   ├── main.o
        │   ├── test_l1_hdmi_cec_driver.o
        │   ├── test_l2_hdmi_cec_sink_driver.o
        │   ├── test_l2_hdmi_cec_source_driver.o
        │   ├── test_l3_hdmi_cec_driver.o
        │   ├── test_register.o
        │   └── test_vcomponent.o
        ├── ut-core
        │   ├── framework
        │   │   └── CUnit-2.1-3
        │   │       └── CUnit
        │   │           └── Sources
        │   │               └── Framework
        │   │                   ├── CUError.o
        │   │                   ├── MyMem.o
        │   │                   ├── TestDB.o
        │   │                   ├── TestRun.o
        │   │                   └── Util.o
        │   └── src
        │       ├── c_source
        │       │   ├── cunit_lgpl
        │       │   │   ├── ut_automated.o
        │       │   │   ├── ut_basic.o
        │       │   │   └── ut_console.o
        │       │   └── ut_cunit.o
        │       ├── ut_kvp_profile.o
        │       └── ut_main.o
        └── weak_stubs
            └── src
                └── hdmi_cec_driver.o


```

kanjoe24 commented:
I have modified the weak_stubs to be a shared lib rather than a static one. Also added tests that implements weak functions for ut_kvp.c and ut_log.c. 

With these changes when ut-test is run with both ut-control and weak_stubs linked following is the result

```shell

Building shared library weak_stubs_libs...
Copy shared library weak_stubs_libs to [/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/bin]
cp /home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/build/linux/lib/libweak_stubs_libs.so /home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/bin
Linking test /home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/linux/obj/ut-test
gcc -ggdb -o0 -Wall /home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/linux/obj/framework/CUnit-2.1-3/CUnit/Sources/Framework/CUError.o /home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/linux/obj/framework/CUnit-2.1-3/CUnit/Sources/Framework/MyMem.o /home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/linux/obj/framework/CUnit-2.1-3/CUnit/Sources/Framework/TestDB.o /home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/linux/obj/framework/CUnit-2.1-3/CUnit/Sources/Framework/TestRun.o /home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/linux/obj/framework/CUnit-2.1-3/CUnit/Sources/Framework/Util.o /home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/linux/obj/src/c_source/cunit_lgpl/ut_automated.o /home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/linux/obj/src/c_source/cunit_lgpl/ut_basic.o /home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/linux/obj/src/c_source/cunit_lgpl/ut_console.o /home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/linux/obj/src/c_source/ut_cunit.o /home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/linux/obj/src/ut_kvp_profile.o /home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/linux/obj/src/ut_main.o /home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/linux/obj/tests/src/c_source/ut_test_assert.o /home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/linux/obj/tests/src/c_source/ut_test_kvp_profile.o /home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/linux/obj/tests/src/ut_test_main.o  -o /home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/linux/obj/ut-test -DUT_CUNIT -DNDEBUG -I/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/include -I/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/../include -I/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/include -I/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/framework/ut-control/include -I/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/src -I/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/framework/CUnit-2.1-3/CUnit/Headers -I/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/src/c_source -I/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/include -I/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/framework/ut-control/include -I/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/src -I/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/framework/CUnit-2.1-3/CUnit/Headers -I/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/src/c_source -D UT_VERSION=\"4.3.0-8-g5b4370e\"   -L/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/framework/ut-control/build/linux/lib -lut_control -Wl,-rpath, -pthread -lpthread -L/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/build/linux/lib -lweak_stubs_libs
make[2]: Leaving directory '/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core'
ut-control LIB Copied to [/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/bin]
make[1]: Leaving directory '/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core'
Copy Assets to [/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/bin/assets] 
Build Complete
make: Leaving directory '/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests'
jpn323@janus ~/workspace/rdk-halif-hdmi_cec/ut/ut-core (feature/gh162-Makefile-changes-for-weak-stubs)$ 
jpn323@janus ~/workspace/rdk-halif-hdmi_cec/ut/ut-core (feature/gh162-Makefile-changes-for-weak-stubs)$ 
jpn323@janus ~/workspace/rdk-halif-hdmi_cec/ut/ut-core (feature/gh162-Makefile-changes-for-weak-stubs)$ 
jpn323@janus ~/workspace/rdk-halif-hdmi_cec/ut/ut-core (feature/gh162-Makefile-changes-for-weak-stubs)$ 
jpn323@janus ~/workspace/rdk-halif-hdmi_cec/ut/ut-core (feature/gh162-Makefile-changes-for-weak-stubs)$ 
jpn323@janus ~/workspace/rdk-halif-hdmi_cec/ut/ut-core (feature/gh162-Makefile-changes-for-weak-stubs)$ 
jpn323@janus ~/workspace/rdk-halif-hdmi_cec/ut/ut-core (feature/gh162-Makefile-changes-for-weak-stubs)$ 
jpn323@janus ~/workspace/rdk-halif-hdmi_cec/ut/ut-core (feature/gh162-Makefile-changes-for-weak-stubs)$ 
jpn323@janus ~/workspace/rdk-halif-hdmi_cec/ut/ut-core (feature/gh162-Makefile-changes-for-weak-stubs)$ cd -
/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/bin
jpn323@janus ~/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/bin (feature/gh162-Makefile-changes-for-weak-stubs)$ 
jpn323@janus ~/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/bin (feature/gh162-Makefile-changes-for-weak-stubs)$ 
jpn323@janus ~/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/bin (feature/gh162-Makefile-changes-for-weak-stubs)$ 
jpn323@janus ~/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/bin (feature/gh162-Makefile-changes-for-weak-stubs)$ 
jpn323@janus ~/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/bin (feature/gh162-Makefile-changes-for-weak-stubs)$ 
jpn323@janus ~/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/bin (feature/gh162-Makefile-changes-for-weak-stubs)$ ls
assets  lib  libut_control.so  libweak_stubs_libs.so  run.sh  ut-test
jpn323@janus ~/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/bin (feature/gh162-Makefile-changes-for-weak-stubs)$ 
jpn323@janus ~/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/bin (feature/gh162-Makefile-changes-for-weak-stubs)$ ./run.sh 
Listing Filename:[/tmp/ut-log_2024-12-20_223849.log-Listing.xml]
Results Filename:[/tmp/ut-log_2024-12-20_223849.log]
Setting Log Path [./logs]
Listing Filename:[./logs/ut-log_2024-12-20_223849.log-Listing.xml]
Results Filename:[./logs/ut-log_2024-12-20_223849.log]
Using Profile[./assets/test_kvp.yaml]

2024-12-20-22:38:49, LOG   , ut_cunit.c,   147 : ---- start of test run ----

 ut-core: Wrapper framework for testing frameworks

***************** CUNIT CONSOLE - MAIN MENU ******************************
(R)un  (S)elect  (L)ist  (A)ctivate  (F)ailures  (O)ptions  (H)elp  (Q)uit
Enter command: r

2024-12-20-22:38:51, LOG   , ut_console.c,   633 : 

2024-12-20-22:38:51, LOG   , ut_console.c,   636 : Running Suite : ut-kvp - assert open / close
2024-12-20-22:38:51, LOG   , ut_console.c,   639 :      Running Test : 'kvp profile open()'
2024-12-20-22:38:51, STEP  , ut_test_kvp_profile.c,   144 : test_ut_kvp_profile_open - start
2024-12-20-22:38:51, STEP  , ut_test_kvp_profile.c,   148 : test_ut_kvp_profile_open - end
2024-12-20-22:38:51, LOG   , ut_console.c,   661 :      Test Complete : 'kvp profile open()'
2024-12-20-22:38:51, LOG   , ut_console.c,   633 : 

2024-12-20-22:38:51, LOG   , ut_console.c,   639 :      Running Test : 'kvp profile getInstance()'
2024-12-20-22:38:51, STEP  , ut_test_kvp_profile.c,   191 : Check ut_kvp_profile_getInstance() - Positive
2024-12-20-22:38:51, LOG   , ut_console.c,   661 :      Test Complete : 'kvp profile getInstance()'
2024-12-20-22:38:51, LOG   , ut_console.c,   633 : 

2024-12-20-22:38:51, LOG   , ut_console.c,   639 :      Running Test : 'kvp profile close()'
2024-12-20-22:38:51, STEP  , ut_test_kvp_profile.c,   153 : test_ut_kvp_profile_close - start
2024-12-20-22:38:51, STEP  , ut_test_kvp_profile.c,   157 : test_ut_kvp_profile_close - end
2024-12-20-22:38:51, LOG   , ut_console.c,   661 :      Test Complete : 'kvp profile close()'
2024-12-20-22:38:51, LOG   , ut_console.c,   633 : 

2024-12-20-22:38:51, LOG   , ut_console.c,   636 : Running Suite : ut-kvp - assert testing yaml 
2024-12-20-22:38:51, LOG   , ut_console.c,   639 :      Running Test : 'kvp profile uint8'
2024-12-20-22:38:51, STEP  , ut_test_kvp_profile.c,    43 : test_ut_kvp_profile_uint8 - start
2024-12-20-22:38:51, STEP  , ut_test_kvp_profile.c,    51 : test_ut_kvp_profile_uint8 - end
2024-12-20-22:38:51, LOG   , ut_console.c,   661 :      Test Complete : 'kvp profile uint8'
2024-12-20-22:38:51, LOG   , ut_console.c,   633 : 

2024-12-20-22:38:51, LOG   , ut_console.c,   639 :      Running Test : 'kvp profile uint16'
2024-12-20-22:38:51, STEP  , ut_test_kvp_profile.c,    58 : test_ut_kvp_profile_uint16 - start
2024-12-20-22:38:51, STEP  , ut_test_kvp_profile.c,    66 : test_ut_kvp_profile_uint16 - end
2024-12-20-22:38:51, LOG   , ut_console.c,   661 :      Test Complete : 'kvp profile uint16'
2024-12-20-22:38:51, LOG   , ut_console.c,   633 : 

2024-12-20-22:38:51, LOG   , ut_console.c,   639 :      Running Test : 'kvp profile uint32'
2024-12-20-22:38:51, STEP  , ut_test_kvp_profile.c,    73 : test_ut_kvp_profile_uint32 - start
2024-12-20-22:38:51, STEP  , ut_test_kvp_profile.c,    81 : test_ut_kvp_profile_uint32 - end
2024-12-20-22:38:51, LOG   , ut_console.c,   661 :      Test Complete : 'kvp profile uint32'
2024-12-20-22:38:51, LOG   , ut_console.c,   633 : 

2024-12-20-22:38:51, LOG   , ut_console.c,   639 :      Running Test : 'kvp profile uint64'
2024-12-20-22:38:51, STEP  , ut_test_kvp_profile.c,    88 : test_ut_kvp_profile_uint64
2024-12-20-22:38:51, STEP  , ut_test_kvp_profile.c,    96 : test_ut_kvp_profile_uint64 - end
2024-12-20-22:38:51, LOG   , ut_console.c,   661 :      Test Complete : 'kvp profile uint64'
2024-12-20-22:38:51, LOG   , ut_console.c,   633 : 

2024-12-20-22:38:51, LOG   , ut_console.c,   639 :      Running Test : 'kvp profile string'
2024-12-20-22:38:51, STEP  , ut_test_kvp_profile.c,   103 : test_ut_kvp_profile_string
2024-12-20-22:38:51, STEP  , ut_test_kvp_profile.c,   110 : test_ut_kvp_profile_string - end
2024-12-20-22:38:51, LOG   , ut_console.c,   661 :      Test Complete : 'kvp profile string'
2024-12-20-22:38:51, LOG   , ut_console.c,   633 : 

2024-12-20-22:38:51, LOG   , ut_console.c,   639 :      Running Test : 'kvp profile bool'
2024-12-20-22:38:51, STEP  , ut_test_kvp_profile.c,   116 : test_ut_kvp_profile_bool
2024-12-20-22:38:51, STEP  , ut_test_kvp_profile.c,   124 : test_ut_kvp_profile_bool - end
2024-12-20-22:38:51, LOG   , ut_console.c,   661 :      Test Complete : 'kvp profile bool'
2024-12-20-22:38:51, LOG   , ut_console.c,   633 : 

2024-12-20-22:38:51, LOG   , ut_console.c,   639 :      Running Test : 'kvp profile list count'
2024-12-20-22:38:51, STEP  , ut_test_kvp_profile.c,   132 : test_ut_kvp_profile_list_count
2024-12-20-22:38:51, STEP  , ut_test_kvp_profile.c,   139 : test_ut_kvp_profile_list_count - end
2024-12-20-22:38:51, LOG   , ut_console.c,   661 :      Test Complete : 'kvp profile list count'
2024-12-20-22:38:51, LOG   , ut_console.c,   633 : 

2024-12-20-22:38:51, LOG   , ut_console.c,   636 : Running Suite : ut-kvp - assert testing json 
2024-12-20-22:38:51, LOG   , ut_console.c,   639 :      Running Test : 'kvp profile uint8'
2024-12-20-22:38:51, STEP  , ut_test_kvp_profile.c,    43 : test_ut_kvp_profile_uint8 - start
2024-12-20-22:38:51, STEP  , ut_test_kvp_profile.c,    51 : test_ut_kvp_profile_uint8 - end
2024-12-20-22:38:51, LOG   , ut_console.c,   661 :      Test Complete : 'kvp profile uint8'
2024-12-20-22:38:51, LOG   , ut_console.c,   633 : 

2024-12-20-22:38:51, LOG   , ut_console.c,   639 :      Running Test : 'kvp profile uint16'
2024-12-20-22:38:51, STEP  , ut_test_kvp_profile.c,    58 : test_ut_kvp_profile_uint16 - start
2024-12-20-22:38:51, STEP  , ut_test_kvp_profile.c,    66 : test_ut_kvp_profile_uint16 - end
2024-12-20-22:38:51, LOG   , ut_console.c,   661 :      Test Complete : 'kvp profile uint16'
2024-12-20-22:38:51, LOG   , ut_console.c,   633 : 

2024-12-20-22:38:51, LOG   , ut_console.c,   639 :      Running Test : 'kvp profile uint32'
2024-12-20-22:38:51, STEP  , ut_test_kvp_profile.c,    73 : test_ut_kvp_profile_uint32 - start
2024-12-20-22:38:51, STEP  , ut_test_kvp_profile.c,    81 : test_ut_kvp_profile_uint32 - end
2024-12-20-22:38:51, LOG   , ut_console.c,   661 :      Test Complete : 'kvp profile uint32'
2024-12-20-22:38:51, LOG   , ut_console.c,   633 : 

2024-12-20-22:38:51, LOG   , ut_console.c,   639 :      Running Test : 'kvp profile uint64'
2024-12-20-22:38:51, STEP  , ut_test_kvp_profile.c,    88 : test_ut_kvp_profile_uint64
2024-12-20-22:38:51, STEP  , ut_test_kvp_profile.c,    96 : test_ut_kvp_profile_uint64 - end
2024-12-20-22:38:51, LOG   , ut_console.c,   661 :      Test Complete : 'kvp profile uint64'
2024-12-20-22:38:51, LOG   , ut_console.c,   633 : 

2024-12-20-22:38:51, LOG   , ut_console.c,   639 :      Running Test : 'kvp profile string'
2024-12-20-22:38:51, STEP  , ut_test_kvp_profile.c,   103 : test_ut_kvp_profile_string
2024-12-20-22:38:51, STEP  , ut_test_kvp_profile.c,   110 : test_ut_kvp_profile_string - end
2024-12-20-22:38:51, LOG   , ut_console.c,   661 :      Test Complete : 'kvp profile string'
2024-12-20-22:38:51, LOG   , ut_console.c,   633 : 

2024-12-20-22:38:51, LOG   , ut_console.c,   639 :      Running Test : 'kvp profile bool'
2024-12-20-22:38:51, STEP  , ut_test_kvp_profile.c,   116 : test_ut_kvp_profile_bool
2024-12-20-22:38:51, STEP  , ut_test_kvp_profile.c,   124 : test_ut_kvp_profile_bool - end
2024-12-20-22:38:51, LOG   , ut_console.c,   661 :      Test Complete : 'kvp profile bool'
2024-12-20-22:38:51, LOG   , ut_console.c,   633 : 

2024-12-20-22:38:51, LOG   , ut_console.c,   639 :      Running Test : 'kvp profile list count'
2024-12-20-22:38:51, STEP  , ut_test_kvp_profile.c,   132 : test_ut_kvp_profile_list_count
2024-12-20-22:38:51, STEP  , ut_test_kvp_profile.c,   139 : test_ut_kvp_profile_list_count - end
2024-12-20-22:38:51, LOG   , ut_console.c,   661 :      Test Complete : 'kvp profile list count'

Run Summary:    Type  Total    Ran Passed Failed Inactive
              suites      3      3    n/a      0        0
               tests     17     17     17      0        0
             asserts     58     58     58      0      n/a

Elapsed time =    0.006 seconds


***************** CUNIT CONSOLE - MAIN MENU ******************************
(R)un  (S)elect  (L)ist  (A)ctivate  (F)ailures  (O)ptions  (H)elp  (Q)uit
Enter command: q

2024-12-20-22:38:54, LOG   , ut_cunit.c,   177 : Logfile:[./logs/ut-log_2024-12-20_223849.log]
2024-12-20-22:38:54, LOG   , ut_cunit.c,   185 : ---- end of test run ----

```

kanjoe24 commented:
On the other hand when ut_control is removed, following is the result


```shell
Building shared library weak_stubs_libs...
Copy shared library weak_stubs_libs to [/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/bin]
cp /home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/build/linux/lib/libweak_stubs_libs.so /home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/bin
Linking test /home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/linux/obj/ut-test
gcc -ggdb -o0 -Wall /home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/linux/obj/framework/CUnit-2.1-3/CUnit/Sources/Framework/CUError.o /home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/linux/obj/framework/CUnit-2.1-3/CUnit/Sources/Framework/MyMem.o /home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/linux/obj/framework/CUnit-2.1-3/CUnit/Sources/Framework/TestDB.o /home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/linux/obj/framework/CUnit-2.1-3/CUnit/Sources/Framework/TestRun.o /home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/linux/obj/framework/CUnit-2.1-3/CUnit/Sources/Framework/Util.o /home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/linux/obj/src/c_source/cunit_lgpl/ut_automated.o /home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/linux/obj/src/c_source/cunit_lgpl/ut_basic.o /home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/linux/obj/src/c_source/cunit_lgpl/ut_console.o /home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/linux/obj/src/c_source/ut_cunit.o /home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/linux/obj/src/ut_kvp_profile.o /home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/linux/obj/src/ut_main.o /home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/linux/obj/tests/src/c_source/ut_test_assert.o /home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/linux/obj/tests/src/c_source/ut_test_kvp_profile.o /home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/linux/obj/tests/src/ut_test_main.o  -o /home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/linux/obj/ut-test -DUT_CUNIT -DNDEBUG -I/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/include -I/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/../include -I/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/include -I/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/framework/ut-control/include -I/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/src -I/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/framework/CUnit-2.1-3/CUnit/Headers -I/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/src/c_source -I/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/include -I/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/framework/ut-control/include -I/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/src -I/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/framework/CUnit-2.1-3/CUnit/Headers -I/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/src/c_source -D UT_VERSION=\"4.3.0-8-g5b4370e\"   -Wl,-rpath, -pthread -lpthread -L/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/build/linux/lib -lweak_stubs_libs
make[2]: Leaving directory '/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core'
ut-control LIB Copied to [/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/bin]
make[1]: Leaving directory '/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core'
Copy Assets to [/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/bin/assets] 
Build Complete
make: Leaving directory '/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests'
jpn323@janus ~/workspace/rdk-halif-hdmi_cec/ut/ut-core (feature/gh162-Makefile-changes-for-weak-stubs)$ 
jpn323@janus ~/workspace/rdk-halif-hdmi_cec/ut/ut-core (feature/gh162-Makefile-changes-for-weak-stubs)$ 
jpn323@janus ~/workspace/rdk-halif-hdmi_cec/ut/ut-core (feature/gh162-Makefile-changes-for-weak-stubs)$ 
jpn323@janus ~/workspace/rdk-halif-hdmi_cec/ut/ut-core (feature/gh162-Makefile-changes-for-weak-stubs)$ cd -
/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/bin
jpn323@janus ~/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/bin (feature/gh162-Makefile-changes-for-weak-stubs)$ ls
assets  lib  libut_control.so  libweak_stubs_libs.so  run.sh  ut-test
jpn323@janus ~/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/bin (feature/gh162-Makefile-changes-for-weak-stubs)$ ./run.sh 
Weak implementation of [UT_log_setLogFilePath]Weak implementation of [UT_log_getLogFilename]Listing Filename:[default_log_file.log-Listing.xml]
Results Filename:[default_log_file.log]
Setting Log Path [./logs]
Weak implementation of [UT_log_setLogFilePath]Weak implementation of [UT_log_getLogFilename]Listing Filename:[default_log_file.log-Listing.xml]
Results Filename:[default_log_file.log]
Using Profile[./assets/test_kvp.yaml]
Weak implementation of [ut_kvp_createInstance]Weak implementation of [ut_kvp_open]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]

 ut-core: Wrapper framework for testing frameworks

***************** CUNIT CONSOLE - MAIN MENU ******************************
(R)un  (S)elect  (L)ist  (A)ctivate  (F)ailures  (O)ptions  (H)elp  (Q)uit
Enter command: r
Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_createInstance]Weak implementation of [ut_kvp_open]Weak implementation of [ut_kvp_createInstance]Weak implementation of [ut_kvp_open]Weak implementation of [ut_kvp_createInstance]Weak implementation of [ut_kvp_open]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_createInstance]Weak implementation of [ut_kvp_open]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getUInt8Field]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getUInt8Field]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getUInt8Field]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getUInt8Field]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getUInt16Field]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getUInt16Field]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getUInt16Field]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getUInt16Field]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getUInt32Field]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getUInt32Field]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getUInt32Field]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getUInt32Field]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getUInt64Field]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getUInt64Field]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getUInt64Field]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getUInt64Field]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getStringField]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getStringField]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getStringField]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getBoolField]Weak implementation of [ut_kvp_getBoolField]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getBoolField]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getBoolField]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getListCount]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getListCount]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getListCount]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_createInstance]Weak implementation of [ut_kvp_open]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getUInt8Field]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getUInt8Field]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getUInt8Field]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getUInt8Field]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getUInt16Field]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getUInt16Field]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getUInt16Field]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getUInt16Field]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getUInt32Field]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getUInt32Field]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getUInt32Field]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getUInt32Field]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getUInt64Field]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getUInt64Field]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getUInt64Field]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getUInt64Field]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getStringField]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getStringField]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getStringField]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getBoolField]Weak implementation of [ut_kvp_getBoolField]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getBoolField]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getBoolField]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getListCount]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getListCount]Weak implementation of [UT_logPrefix]Weak implementation of [ut_kvp_getListCount]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]

Run Summary:    Type  Total    Ran Passed Failed Inactive
              suites      3      3    n/a      0        0
               tests     17     17      2     15        0
             asserts     57     57      8     49      n/a

Elapsed time =    0.001 seconds


***************** CUNIT CONSOLE - MAIN MENU ******************************
(R)un  (S)elect  (L)ist  (A)ctivate  (F)ailures  (O)ptions  (H)elp  (Q)uit
Enter command: q
Weak implementation of [UT_log_getLogFilename]Weak implementation of [UT_logPrefix]Weak implementation of [UT_logPrefix]jpn323@janus ~/workspace/rdk-halif-hdmi_cec/ut/ut-core/tests/build/bin (feature/gh162-Makefile-changes-for-weak-stubs)$ 

```

kanjoe24 commented:
Also the directory structure from hal interface looks like below

```shell
 tree ut/build
ut/build
└── linux
    ├── lib
    │   └── libweak_stubs_libs.so
    └── obj
        ├── hal_test
        ├── src
        │   ├── main.o
        │   ├── test_l1_hdmi_cec_driver.o
        │   ├── test_l2_hdmi_cec_sink_driver.o
        │   ├── test_l2_hdmi_cec_source_driver.o
        │   ├── test_l3_hdmi_cec_driver.o
        │   ├── test_register.o
        │   └── test_vcomponent.o
        ├── ut-core
        │   ├── framework
        │   │   └── CUnit-2.1-3
        │   │       └── CUnit
        │   │           └── Sources
        │   │               └── Framework
        │   │                   ├── CUError.o
        │   │                   ├── MyMem.o
        │   │                   ├── TestDB.o
        │   │                   ├── TestRun.o
        │   │                   └── Util.o
        │   └── src
        │       ├── c_source
        │       │   ├── cunit_lgpl
        │       │   │   ├── ut_automated.o
        │       │   │   ├── ut_basic.o
        │       │   │   └── ut_console.o
        │       │   └── ut_cunit.o
        │       ├── ut_kvp_profile.o
        │       └── ut_main.o
        └── weak_stubs
            └── src
                └── hdmi_cec_driver.o

15 directories, 21 files
jpn323@janus ~/workspace/rdk-halif-hdmi_cec (develop)$ tree ut/bin/
ut/bin/
├── hal_test
├── lib
│   └── libweak_stubs_libs.so
├── libut_control.so
└── libweak_stubs_libs.so
```

Ulrond commented on Makefile:
```diff
@@ -75,6 +75,18 @@ else # GTEST case
   SRCS := $(shell find $(SRC_DIRS) -type f \( -name '*.cpp' -o -name '*.c' \) | grep -v "$(EXCLUDE_DIRS)")
 endif
 
+# Check if BUILD_WEAK_STUBS_SRC is defined
```
you need a script now to trigger and automatically call the tests..

And checking whether all the tests pass or not.

kanjoe24 replied:
Have added ut_test_scripts.sh which will check this scenario.

avyak123 commented:
Hi,
Tried cross compiling bridge-util hal with the changes in the latest commit, and it worked as expected. The weak symbols from the skeleton shared library were successfully overridden by the strong symbols from the vendor-specific library when the hal test binary was executed. 
Thanks

Ulrond commented on Makefile:
```diff
@@ -75,6 +75,18 @@ else # GTEST case
+# Check if BUILD_WEAK_STUBS_SRC is defined
+ifdef BUILD_WEAK_STUBS_SRC
+
+# Define variables for the static library and its source files
```
upgrade the README.md to support this section

---

### Feature: `BUILD_WEAK_STUBS_SRC` for Weak Library Compilation

The `BUILD_WEAK_STUBS_SRC` variable enables clients to define and export a list of source files that will be compiled into a weak library. This allows testing suites to use stubbed implementations of functions during development, with the strong implementation provided by vendors or third parties at a later stage.

#### Benefits:
1. **Supports Testing Development**: Enables testing suites to run without waiting for strong implementations.
2. **Decouples Development**: Testing can proceed independently of vendor timelines.
3. **Seamless Replacement**: Strong implementations can replace weak stubs without changes to the testing suites.

#### Key Makefile Logic:
```make
WEAK_STUBS_LIB := $(LIB_DIR)/libweak_stubs_libs.so
WEAK_STUBS_OUTPUT_DIR ?= $(BUILD_DIR)/weak_stubs/src
WEAK_STUBS_SRC := $(shell find $(BUILD_WEAK_STUBS_SRC) -name *.c)
WEAK_STUBS_OBJ := $(patsubst $(BUILD_WEAK_STUBS_SRC)/%, $(WEAK_STUBS_OUTPUT_DIR)/%, $(WEAK_STUBS_SRC:.c=.o))
```
- `BUILD_WEAK_STUBS_SRC`: Path to source files for weak stubs.
- `WEAK_STUBS_LIB`: Compiled weak library (`libweak_stubs_libs.so`).
- `WEAK_STUBS_SRC` and `WEAK_STUBS_OBJ`: Dynamically map source to object files.

#### Workflow:
1. **Define Stubs**: Clients export `BUILD_WEAK_STUBS_SRC` with the path to stub source files.
   ```bash
   export BUILD_WEAK_STUBS_SRC=/path/to/stubs
   ```
2. **Build Library**: The Makefile compiles the stubs into a weak library.
3. **Develop and Test**: Testing suites use the weak library.
4. **Integrate Strong Implementation**: Vendors replace the weak stubs when ready.

This approach ensures smooth testing suite development while awaiting third-party components.

kanjoe24 replied:
Done.

kanjoe24 commented:
Release script release:

```shell
./release-test-script-ut-core.sh -t feature/gh162-Makefile-changes-for-weak-stubs
UT_CORE_BRANCH_NAME = feature/gh162-Makefile-changes-for-weak-stubs
UT_CONTROL_BRANCH_NAME = 
UT_CONTROL_BRANCH_NAME is empty
==========================================================
RESULTS for ubuntu 
On the branch feature/gh162-Makefile-changes-for-weak-stubs
. PASS
framework/ut-control/build/linux/curl/lib/libcurl.a exists. PASS
Openssl static lib does not exist. PASS 
CMake host binary does not exist. PASS 
build/linux/cpp_libs/lib/libgtest.a exists. PASS
build/linux/cpp_libs/lib/libgtest_main.a exists. PASS
==========================================================
==========================================================
RESULTS for VM-SYNC 
On the branch feature/gh162-Makefile-changes-for-weak-stubs
. PASS
framework/ut-control/build/linux/curl/lib/libcurl.a exists. PASS
framework/ut-control/build/linux/openssl/lib/libssl.a exists. PASS 
framework/ut-control/host-tools/CMake-3.30.0/build/bin/cmake exists. PASS 
build/linux/cpp_libs/lib/libgtest.a exists. PASS
build/linux/cpp_libs/lib/libgtest_main.a exists. PASS
==========================================================
==========================================================
RESULTS for dunfell_arm 
On the branch feature/gh162-Makefile-changes-for-weak-stubs
. PASS
framework/ut-control/build/arm/curl/lib/libcurl.a exists. PASS
framework/ut-control/build/arm/openssl/lib/libssl.a exists. PASS 
CMake host binary does not exist. PASS 
build/arm/cpp_libs/lib/libgtest.a exists. PASS
build/arm/cpp_libs/lib/libgtest_main.a exists. PASS
==========================================================
==========================================================
RESULTS for dunfell_linux 
On the branch feature/gh162-Makefile-changes-for-weak-stubs
. PASS
framework/ut-control/build/linux/curl/lib/libcurl.a exists. PASS
Openssl static lib does not exist. PASS 
CMake host binary does not exist. PASS 
build/linux/cpp_libs/lib/libgtest.a exists. PASS
build/linux/cpp_libs/lib/libgtest_main.a exists. PASS
==========================================================
==========================================================
RESULTS for kirkstone_arm 
On the branch feature/gh162-Makefile-changes-for-weak-stubs
. PASS
framework/ut-control/build/arm/curl/lib/libcurl.a exists. PASS
framework/ut-control/build/arm/openssl/lib/libssl.a exists. PASS 
CMake host binary does not exist. PASS 
build/arm/cpp_libs/lib/libgtest.a exists. PASS
build/arm/cpp_libs/lib/libgtest_main.a exists. PASS
==========================================================
==========================================================
RESULTS for kirkstone_linux 
On the branch feature/gh162-Makefile-changes-for-weak-stubs
. PASS
framework/ut-control/build/linux/curl/lib/libcurl.a exists. PASS
Openssl static lib does not exist. PASS 
CMake host binary does not exist. PASS 
build/linux/cpp_libs/lib/libgtest.a exists. PASS
build/linux/cpp_libs/lib/libgtest_main.a exists. PASS
==========================================================

```

kanjoe24 commented:
Release script results for halif platform:

```shell
 ./release-test-script-platform.sh -t feature/gh162-Makefile-changes-for-weak-stubs
UT_CORE_BRANCH_NAME = feature/gh162-Makefile-changes-for-weak-stubs
UT_CONTROL_BRANCH_NAME = 
UT_CONTROL_BRANCH_NAME is empty
==========================================================
RESULTS for ubuntu 
On the branch feature/gh162-Makefile-changes-for-weak-stubs
            . PASS
/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/scripts/rdk-halif-hdmi_cec-ubuntu
ut/bin/hal_test exists. PASS
ut/ut-core/framework/ut-control/build/linux/curl/lib/libcurl.a exists. PASS
Openssl static lib does not exist. PASS 
CMake host binary does not exist. PASS 
==========================================================
==========================================================
RESULTS for VM-SYNC 
On the branch feature/gh162-Makefile-changes-for-weak-stubs
            . PASS
/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/scripts/rdk-halif-hdmi_cec-VM-SYNC
ut/bin/hal_test exists. PASS
ut/ut-core/framework/ut-control/build/linux/curl/lib/libcurl.a exists. PASS
ut/ut-core/framework/ut-control/build/linux/openssl/lib/libssl.a exists. PASS 
ut/ut-core/framework/ut-control/host-tools/CMake-3.30.0/build/bin/cmake exists. PASS 
==========================================================
==========================================================
RESULTS for dunfell_arm 
On the branch feature/gh162-Makefile-changes-for-weak-stubs
            . PASS
/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/scripts/rdk-halif-hdmi_cec-dunfell_arm
ut/bin/hal_test exists. PASS
ut/ut-core/framework/ut-control/build/arm/curl/lib/libcurl.a exists. PASS
ut/ut-core/framework/ut-control/build/arm/openssl/lib/libssl.a exists. PASS 
CMake host binary does not exist. PASS 
==========================================================
==========================================================
RESULTS for dunfell_linux 
On the branch feature/gh162-Makefile-changes-for-weak-stubs
            . PASS
/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/scripts/rdk-halif-hdmi_cec-dunfell_linux
ut/bin/hal_test exists. PASS
CURL static lib does not exist. PASS 
Openssl static lib does not exist. PASS 
CMake host binary does not exist. PASS 
==========================================================
==========================================================
RESULTS for kirkstone_arm 
On the branch feature/gh162-Makefile-changes-for-weak-stubs
            . PASS
/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/scripts/rdk-halif-hdmi_cec-kirkstone_arm
ut/bin/hal_test exists. PASS
ut/ut-core/framework/ut-control/build/arm/curl/lib/libcurl.a exists. PASS
ut/ut-core/framework/ut-control/build/arm/openssl/lib/libssl.a exists. PASS 
CMake host binary does not exist. PASS 
==========================================================
==========================================================
RESULTS for kirkstone_linux 
On the branch feature/gh162-Makefile-changes-for-weak-stubs
            . PASS
/home/jpn323/workspace/rdk-halif-hdmi_cec/ut/ut-core/scripts/rdk-halif-hdmi_cec-kirkstone_linux
ut/bin/hal_test exists. PASS
ut/ut-core/framework/ut-control/build/linux/curl/lib/libcurl.a exists. PASS
Openssl static lib does not exist. PASS 
CMake host binary does not exist. PASS 
==========================================================

```
